src/OpenLoco/src/Paint/PaintTile.cpp

```diff
@@ -97,10 +99,122 @@ namespace OpenLoco::Paint
 
+    constexpr std::array<std::array<World::Pos3, 9>, 2> kSupportBoundingBoxOffsets = {
+        std::array<World::Pos3, 9>{
+            World::Pos3{ 2, 2, 6 },
+            World::Pos3{ 28, 2, 6 },
+            World::Pos3{ 2, 28, 6 },
+            World::Pos3{ 28, 28, 6 },
+            World::Pos3{ 15, 15, 6 },
+            World::Pos3{ 15, 2, 6 },
+            World::Pos3{ 2, 15, 6 },
+            World::Pos3{ 28, 15, 6 },
+            World::Pos3{ 15, 28, 6 },
+        },
+        std::array<World::Pos3, 9>{
+            World::Pos3{ 2, 2, 28 },
+            World::Pos3{ 28, 2, 28 },
+            World::Pos3{ 2, 28, 28 },
+            World::Pos3{ 28, 28, 28 },
+            World::Pos3{ 15, 15, 28 },
+            World::Pos3{ 15, 2, 28 },
+            World::Pos3{ 2, 15, 28 },
+            World::Pos3{ 28, 15, 28 },
+            World::Pos3{ 15, 28, 28 },
+        },
+    };
+
+    constexpr std::array<World::Pos3, 2> kSupportBoundingBoxLengths = {
+        World::Pos3{ 1, 1, 17 },
+        World::Pos3{ 1, 1, 1 },
+    };
+
+    constexpr std::array<std::array<uint8_t, 4>, 4> kFrequencyRotationMap = {
+        std::array<uint8_t, 4>{ 1U << 0, 1U << 2, 1U << 1, 1U << 3 },
+        std::array<uint8_t, 4>{ 1U << 3, 1U << 0, 1U << 2, 1U << 1 },
+        std::array<uint8_t, 4>{ 1U << 1, 1U << 3, 1U << 0, 1U << 2 },
+        std::array<uint8_t, 4>{ 1U << 2, 1U << 1, 1U << 3, 1U << 0 },
+    };
+
     // 0x0046748F
-    static void sub_46748F([[maybe_unused]] PaintSession& session)
+    static void paintSupports(PaintSession& session)
     {
-        call(0x0046748F);
+        // Copy the supports
+        const TrackRoadAdditionSupports supports = session.getAdditionSupport();
+        // Clear the supports as this function will have taken care of their render
+        session.setAdditionSupport(TrackRoadAdditionSupports{});
+
+        auto& bridge = session.getBridgeEntry();
+        if (!bridge.isEmpty())
+        {
+            auto* bridgeObj = ObjectManager::get<BridgeObject>(bridge.objectId);
+            // Bridge blocks the supports due to the roof
+            if (bridgeObj->noRoof & (1U << 0))
+            {
+                return;
+            }
+        }
+
+        const auto pos = session.getSpritePosition();
+        for (auto i = 0U; i < std::size(kSegmentOffsets); ++i)
+        {
+            const auto seg = kSegmentOffsets[i];
+
+            // No support at this location
+            if (supports.segmentImages[i] == 0)
+            {
+                continue;
+            }
+            // Support blocked by something at this location
+            if ((supports.occupiedSegments & seg) != SegmentFlags::none)
+            {
+                continue;
+            }
+
+            const auto frequency = supports.segmentFrequency[i];
+
+            bool frequenceSkip = [&]() {
+                auto& line = kFrequencyRotationMap[session.getRotation()];
+
+                if ((frequency & line[0]) && !(pos.x & 0b0010'0000))
```

Comment:
Is this still correct? The previous code had different conditions per rotation, 2 and 3 required the second condition to be true instead of false.

Reply:
I'm pretty sure it's the same. What i did to test was turn the old code into a lambda as well then assert the old and new were the same. It's not completely foolproof as i can't be sure every code path was exercised.

Reply:
Just checked, you are right.